centreon/src/Core/Host/Application/UseCase/PartialUpdateHost/PartialUpdateHost.php

```diff
@@ -0,0 +1,611 @@
+                )
+                || ! ($host = $this->readHostRepository->findById($hostId))
+            ) {
+                $this->error(
+                    'Host not found',
+                    ['host_id' => $hostId]
+                );
+                $presenter->setResponseStatus(new NotFoundResponse('Host'));
+
+                return;
+            }
+
+            $this->updatePropertiesInTransaction($request, $host);
+
+            $presenter->setResponseStatus(new NoContentResponse());
+        } catch (HostException $ex) {
+            $presenter->setResponseStatus(
+                match ($ex->getCode()) {
+                    HostException::CODE_CONFLICT => new ConflictResponse($ex),
+                    default => new ErrorResponse($ex),
+                }
+            );
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+        } catch (\Throwable $ex) {
+            $presenter->setResponseStatus(new ErrorResponse(HostException::editHost()));
+            $this->error((string) $ex);
+        }
+    }
+
+    private function updatePropertiesInTransaction(PartialUpdateHostRequest $request, Host $host): void
+    {
+        try {
+            $this->dataStorageEngine->startTransaction();
+
+            $previousMonitoringServer = $host->getMonitoringServerId();
+            $this->updateHost($request, $host);
+            $this->updateHostCategories($request, $host);
+            $this->updateHostGroups($request, $host);
+            $this->updateParentTemplates($request, $host);
+            // Note: parent templates must be updated before macros for macro inheritance resolution
+            $this->updateMacros($request, $host);
+
+            $this->writeMonitoringServerRepository->notifyConfigurationChange($host->getMonitoringServerId());
+            if ($previousMonitoringServer !== $host->getMonitoringServerId()) {
+                // Monitoring server has change, notify previous monitoring server of configuration changes.
+                 $this->writeMonitoringServerRepository->notifyConfigurationChange($previousMonitoringServer);
+            }
+
+            $this->dataStorageEngine->commitTransaction();
+        } catch (\Throwable $ex) {
+            $this->error("Rollback of 'PartialUpdateHost' transaction.");
+            $this->dataStorageEngine->rollbackTransaction();
+
+            throw $ex;
+        }
+    }
+
+    /**
+     * @param PartialUpdateHostRequest $dto
+     * @param Host $host
+     *
+     * @throws \Throwable|AssertionFailedException|HostException
+     */
+    private function updateHost(PartialUpdateHostRequest $dto, Host $host): void
+    {
+        $this->info('PartialUpdateHost: update host', ['host_id' => $host->getId()]);
+
+        $inheritanceMode = $this->optionService->findSelectedOptions(['inheritance_mode']);
+        $inheritanceMode = isset($inheritanceMode[0])
+            ? (int) $inheritanceMode[0]->getValue()
+            : 0;
```

Comment:
What does the value 0 mean for inheritance mode?

Reply:
I'll change it to null